multistep/commonsteps/step_download.go

```diff
@@ -56,12 +57,39 @@ type StepDownload struct {
 }
 
 var defaultGetterClient = getter.Client{
-	Getters: getter.Getters,
-}
-
-func init() {
-	defaultGetterClient.Getters = append(defaultGetterClient.Getters, new(gcs.Getter))
-	defaultGetterClient.Getters = append(defaultGetterClient.Getters, new(s3.Getter))
+	// Disable writing and reading through symlinks
+	DisableSymlinks: true,
+	// The order of the Getters in the list may affect the result
+	// depending if the Request.Src is detected as valid by multiple getters
+	Getters: []getter.Getter{
+		&getter.GitGetter{
+			Timeout: 5 * time.Minute,
+			Detectors: []getter.Detector{
+				new(getter.GitHubDetector),
+				new(getter.GitDetector),
+				new(getter.BitBucketDetector),
+				new(getter.GitLabDetector),
+			},
+		},
+		&getter.HgGetter{
+			Timeout: 5 * time.Minute,
+		},
+		new(getter.SmbClientGetter),
+		new(getter.SmbMountGetter),
+		&getter.HttpGetter{
+			Netrc:                 true,
+			XTerraformGetDisabled: true,
+			HeadFirstTimeout:      10 * time.Second,
+			ReadTimeout:           30 * time.Second,
```

Comment:
The other getter timeouts are 5 minutes, while this one is essentially 30 seconds. Maybe it'd make more sense to have a uniform timeout across the getters, or somehow make these configurable?

Reply:
I like the idea of the uniform timeout. TBH 30 seconds seems really short given the use of go-getter in Packer. I struggled a bit to find good defaults. But ultimately think that going high 30 minutes is a good base line. We can make these configurable for the step so that plugins could override if needed. 

That said there is an open issue for improving iso_config to support some addition go-getter Headers, which could also include setting some of these values. Taking a  page out of Nomad's book here.